inventory.py

```diff
@@ -167,11 +175,112 @@ def check_if_valid_user_id(id: int, user_ids: Sequence[int]) -> bool:
+    :param response:
+    :type response: requests
+    :return: next_page_url
+    :rtype: str
+    """
+    logging.debug(get_next_page_url.__name__ + '() called')
+    results = response.result().links
+    if not results:
+        logging.debug('The api call do not have Link headers')
+        return None
+
+    if 'next' in results:
+        url_ = results['next']['url']
+        published_course_next_page_list.append(url_)
+        logger.info(f"Pagination size {len(published_course_next_page_list)} for published_at date")
+
+
+def published_date_resp_parsing(response):
+    logger.info("published_date_resp_parsing Call")
+
+    if response is None:
+        logger.info(f"Published course date response is None ")
+        return
+
+    start_time = time.time()
+
+    logger.info(f"published courses date collected so far : {len(published_course_date)}")
+    status = response.result().status_code
+    published_date_found = False
+    if status != 200:
+        logger.info(f"Response not successful with status code {status} due to {response.result().text}")
+        return
+
+    try:
+        audit_events = json.loads(response.result().text)
+    except JSONDecodeError as e:
+        logger.info(f"Error in parsing the response {e.message}")
+        return
+
+    if not audit_events:
+        logger.info(f"Response for fetching published date is empty {audit_events}")
+        return
+
+    events = audit_events['events']
+
+    for event in events:
+        if event['event_type'] == 'published':
+            course_id = event['links']['course']
+            published_date_found = True
+            published_course_date.update({course_id: event['created_at']})
+            logger.info(f"Published Date {event['created_at']} for course {course_id}")
+            break
+    if not published_date_found:
+        get_next_page_url(response)
+
+    seconds = time.time() - start_time
+    str_time = time.strftime("%H:%M:%S", time.gmtime(seconds))
+    logger.debug(f"Parsing the published date took {str_time} ")
+    return
+
+
+def get_published_course_date(course_ids, next_page_links=None):
+    logger.info("Starting of get_published_course_date call")
+    with FuturesSession(max_workers=NUM_ASYNC_WORKERS) as session:
+        headers = {'Content-type': 'application/json', 'Authorization': 'Bearer ' + CANVAS_TOKEN}
+        if next_page_links is not None:
+            logger.info("Going through Next page URL set")
+            responses = []
+            for next_page_link in next_page_links:
+                response = session.get(next_page_link, headers=headers)
+                published_course_next_page_list.remove(next_page_link)
+                responses.append(response)
+        else:
+            logger.info("Initial Round of Fetching course published date")
+            responses = [
+                session.get(f'{CANVAS_URL}/api/v1/audit/course/courses/{course_id}?per_page=100', headers=headers)
```

Comment:
Does `session.get` accept query parameters? I'm wondering why we're hardcoding the `per_page` parameter here.